dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/container/EJSContainer.java

```diff
@@ -2576,6 +2578,20 @@ public void preInvokeMdbAfterActivate(EJSWrapperBase wrapper,
         // the bean is activated.
         preInvokeAfterActivate(wrapper, eb, s, args);
 
+        // If there is a registered message endpoint collaborator, call it for preInvoke processing.
+        MessageEndpointCollaborator meCollaborator = getEJBRuntime().getMessageEndpointCollaborator(wrapper.bmd);
+        if (meCollaborator != null) {
+            HashMap<String, Object> contextData = new HashMap<String, Object>();
+            contextData.put(MessageEndpointCollaborator.KEY_ACTIVATION_SPEC_ID, wrapper.bmd.ivActivationSpecJndiName);
+            contextData.put(MessageEndpointCollaborator.KEY_J2EE_NAME, wrapper.bmd.j2eeName);
+
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "Invoking MECollaborator " + meCollaborator + " for preInvoke processing with the following context data: " + contextData);
+            }
+            
+            s.messageEndpointContext = meCollaborator.preInvoke(contextData);
```

Comment:
I guess this is a nit, but is there a reason we are setting the messageEndpointContext directly instead of calling `s.setMessageEndpointContext(meCollaborator.preInvoke(contextData));`?  Someone searching for calls to `setMessageEndpointContext()` (like me) wouldn't find it the way it is now.

Reply:
This is actually consistent with the rest of EJSContainer. For performance, the EJB core code generally accesses variables for collaborators directly.  Only the EJB code in other projects use the set/get methods since they don't have visibility to the variables.  I'm fine with the code as is.